.travis.yml

```diff
@@ -22,62 +22,98 @@ matrix:
             - secure: "S3u2VCE2Vy8KNXoeh+DhnzjCmgTX0r95uEZrXDU+IKANOOCKn7Dg4OFDZE3LY/i1y2/EUDpnR5yLC38Ks795EUP/sv/OoMl4tjQ20yERjqWh+gcIRrgx7SdVabuAh3t4aBdaLD4Pfnj5avxeCt6rL7yGnj0wdbrbJSBZPsgSnuQ="
           after_success:
-            # Android jobs are triggered from cron and overwrite `after_sucess` part
-            - if [[ "$TRAVIS_EVENT_TYPE" == "cron" ]] ; then exit 0 ; fi
-            - sudo chown -R $USER build
-            - cd build
+            - |
+              set -e
+              if [[ "$TRAVIS_EVENT_TYPE" == "cron" ]]; then
+                # Android jobs are triggered from cron and overwrite `after_sucess` part
+                exit 0
+              fi
+              sudo chown -R $USER build
+              cd build
               # make install is done inside docker
-            - mv OpenRCT2/bin/openrct2 OpenRCT2/ && mv OpenRCT2/share/openrct2 OpenRCT2/data && mv OpenRCT2/share/doc/openrct2 OpenRCT2/doc
-            - rm -rf OpenRCT2/bin OpenRCT2/share # remove empty dirs
-            - tar cvzf openrct2-linux.tar.gz OpenRCT2/
-            - if [[ "z${TRAVIS_TAG}" != "z" ]] ; then
-              export PUSH_BRANCH=master ;
-              else export PUSH_BRANCH=$TRAVIS_BRANCH ; export FILENAME_PART=-${TRAVIS_BRANCH}-$(git rev-parse --short HEAD) ;
+              file=$(find -maxdepth 1 -name "openrct2-*.tar.gz")
+              full_file=$(readlink -m ${file})
+              filename=$(basename ${full_file})
+              if [[ "z${TRAVIS_TAG}" != "z" ]]; then
+                export PUSH_BRANCH=master
+              else
+                export PUSH_BRANCH=$TRAVIS_BRANCH
+                export FILENAME_PART=-${TRAVIS_BRANCH}-$(git rev-parse --short HEAD)
               fi
-            - if [[ "z$OPENRCT2_ORG_TOKEN" != "z" && "$TRAVIS_PULL_REQUEST" == "false" && ("${TRAVIS_BRANCH}" =~ ^(develop|push/) || "z${TRAVIS_TAG}" != "z") ]] ; then
-              curl -o - -v --form "key=$OPENRCT2_ORG_TOKEN" --form "fileName=OpenRCT2-${OPENRCT2_VERSION}${FILENAME_PART}-linux-i686.tar.gz" --form "version=${OPENRCT2_VERSION}" --form "gitHash=$TRAVIS_COMMIT" --form "gitBranch=$PUSH_BRANCH" --form "flavourId=4" --form "file=@openrct2-linux.tar.gz" "https://openrct2.org/altapi/?command=push-build";
-              else curl --progress-bar --upload-file openrct2-linux.tar.gz https://transfer.sh/openrct2-linux-i686.tar.gz -o link && cat link;
+              if [[ "z$OPENRCT2_ORG_TOKEN" != "z" && "$TRAVIS_PULL_REQUEST" == "false" && ("${TRAVIS_BRANCH}" =~ ^(develop|push/) || "z${TRAVIS_TAG}" != "z") ]]; then
+                curl -o - \
+                  -v \
+                  --form "key=$OPENRCT2_ORG_TOKEN" \
+                  --form "fileName=OpenRCT2-${OPENRCT2_VERSION}${FILENAME_PART}-linux-i686.tar.gz" \
+                  --form "version=${OPENRCT2_VERSION}" \
+                  --form "gitHash=$TRAVIS_COMMIT" \
+                  --form "gitBranch=$PUSH_BRANCH" \
+                  --form "flavourId=9" \
```

Comment:
4.

This part is almost the same for amd64 and i686 jobs, maybe you can extract it to a script? We could then reuse it for Android's APKs too.